index.js

```diff
@@ -25,15 +26,34 @@ module.exports = (file, opts) => {
 		return Promise.reject(new Error('Size must be 1024 or less'));
 	}
 
-	if (opts.destination && typeof opts.destination !== 'string') {
-		return Promise.reject(new Error(`Expected \`destination\` to be of type \`string\`, got \`${typeof opts.destination}\``));
+	return execa.stdout(bin, [file, opts.size], {encoding: 'buffer'});
+};
+
+exports.file = (file, opts) => {
```

Comment:
I've started a new pattern. By wrapping the whole thing in a promise I can be absolute sure nothing throws synchronously and I don't have to have all the verbose `return Promise.reject()` stuff.

`exports.file = (file, opts) => {` => `exports.file = (file, opts) => Promise.resolve().then({`

When we support async functions, we can simply remove the wrapper and it will work fine.